### PR TITLE
Deep Wide Selu Neural Network

### PR DESCRIPTION
**Pull request by bluejay9676**
and indices
- Added Tensorflow NN model
- General parameters to run the model.
- Needs the data to format the input.
- Need to freeze the learned weights (will have to search that up)

### CODE REVIEW

File: index.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## NN paramters
```
Comment (riceluxs1t) [inline index.py]: 여기 정의 된 상수들 constants.py 로 옮기는게 어떨까

File: index.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+learning_rate = 0.001
+training_epochs = 500
+keep_rate = 0.7
```
Comment (riceluxs1t) [inline index.py]: 그리고 주로 상수값들은 대문자로 LEARNING_RATE 이렇게 하는게 convention

File: NN.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import numbers
+from tensorflow.contrib import layers
+from tensorflow.python.framework import ops
+from tensorflow.python.framework import tensor_shape
+from tensorflow.python.framework import tensor_util
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import random_ops
+from tensorflow.python.ops import array_ops
+from tensorflow.python.layers import utils
+
```
Comment (riceluxs1t) [inline NN.py]: 마이너한 nit 인데 주로 같은 파일에서 불러온 함수들은 같은 라인에 적는게 좋아.
from tensorflow.python.ops import math_ops, random_ops, array_ops

또한, import 순서는 주로 외부 3rd party module 들을 alphanumerical 한 순서대로
그리고 한줄 띄우고 내부 module 들 alphanumerical 한 순서대로 정렬함.

File: NN.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+tf.set_random_seed(777)  # reproducibility
```
Comment (riceluxs1t) [inline NN.py]: run 을 하는 runner 같은애를 따로 분리해서 걔가 이런 값들을 실행하도록 하는게 좋을듯

File: NN.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+def selu(x):
+    with ops.name_scope('elu') as scope:
+        alpha = 1.6732632423543772848170429916717
+        scale = 1.0507009873554804934193349852946
+        return scale*tf.where(x>=0.0, x, alpha*tf.nn.elu(x))
+
+
+def dropout_selu(x, keep_prob, alpha= -1.7580993408473766, fixedPointMean=0.0, fixedPointVar=1.0,
+                 noise_shape=None, seed=None, name=None, training=False):
+    """Dropout to a value with rescaling."""
```
Comment (riceluxs1t) [inline NN.py]: def dropout_selu(
     x,
     keep_prob,
     alpha=....
) 
로 정리

Comment (riceluxs1t) [reply NN.py]: x, keep_prob, alpha 앞에 탭 넣어서

File: NN.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+        ret = a * ret + b
+        ret.set_shape(x.get_shape())
+        return ret
+
+    with ops.name_scope(name, "dropout", [x]) as name:
+        return utils.smart_cond(training,
+                                lambda: dropout_selu_impl(x, keep_prob, alpha, noise_shape, seed, name),
+                                lambda: array_ops.identity(x))
+
+# DATA
```
Comment (riceluxs1t) [inline NN.py]: 이 아래 실행하는 파트들도 들도 다 runner 같은거로 분리.

File: index.py (hunk 2 above)
Comment (riceluxs1t) [inline index.py]: 그리고 파일 이름은 NN.py 보다 model.py 로

**Comment by bluejay9676** [comment]
- Runner & Model separated
- Parameters in index.py -> constants.py
- Fixed according to the coding convention

**Comment by bluejay9676** [comment]
Accuracy of 73% model

File: runner.py [hunk 7]
```diff
@@ -0,0 +1,138 @@
+"""
```
Comment (riceluxs1t) [inline runner.py:1]: this file is just written vey terribly.
1) the way it is written, the expensive modeling part is executed every single time this file is imported. 
2) Some hyper parameters  that determine which data to use, what to name the output file, should all be input from the user.

Solutions,
1) wrap the executable part in 

if __name__ == '__main__':

code block,
this ensures that all the lines (not wrapped in functions, classes) are only run when you run something like python runner.py ......arguments.... from your terminal

2) AVOID HARD CODING. engineering principle 101. have them be input from the user instead. the standard thing to use here is to use python's builtin argparse module. This wraps stuff around the vanilla input(), raw_input() stuff and makes your life easier to write python executables. example code https://gist.github.com/riceluxs1t/60cb8509d9dc7bcecacf335120e3386c

File: constants.py [hunk 8]
```diff
@@ -31,3 +31,12 @@
     '고척': 'GoChuk',
     '수원': 'SuWon'
 }
+
+## NN paramters
```
Comment (riceluxs1t) [inline constants.py:35]: Have all these be user input.

File: model.py [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+from tensorflow.python.framework import ops, tensor_shape, tensor_util
+from tensorflow.python.ops import math_ops, random_ops, array_ops
+from tensorflow.python.layers import utils
+
+from constants import *
+
+
+def selu(x):
+    with ops.name_scope('elu') as scope:
+        alpha = 1.6732632423543772848170429916717
```
Comment (riceluxs1t) [inline model.py:19]: these should be constants.

**Review by riceluxs1t — CHANGES_REQUESTED** [review]
